lib/ContentType/RestFieldsRegistry.php

```diff
@@ -48,13 +48,21 @@ static protected function get_rest_from_post( array $post, array $objects ) : ar
 
 			foreach ( $fields as $field ) {
 
-				// Check first if a custom getter was defined for a given field
+				// Check first if a custom getter was defined for a given field.
 				if ( is_callable( [ $obj, "get_rest_{$field}" ] ) ) {
 					$data[ $field ] = call_user_func( [ $obj, "get_rest_{$field}" ] );
 					continue;
 				}
 
-				$data[ $field ] = get_field( $field, $post['id'] );
+				// If ACF detected, use it to retrieve metadata.
```

Comment:
This is not something I believe you should be doing. You should assume WordPress meta fetching and naming structure. ACF requires you to be inside that mindset, here you are providing a sensible default, but for the fields that don't make sense using such a default, you have the custom callbacks, which put the strain on the developers to be aware of this.

Essentially you stop thinking about ACF, but add to the documentation how to handle complex types in it. Say a repeater, which generates lot's of meta fields, and whose main field only returns the number of entries, you would advise developers to use the `get_field()` as their callback, and not worry about it.